### PR TITLE
feat(task-uuid): Update identifier of Task from task_id to uuid

### PR DESCRIPTION
**Pull request by jatinriverlane**
# Changes
* Made relevant changes to update `task_id` to `uuid`
* Improved `humanize_list` utility function

### CODE REVIEW

**Review by str-anger — CHANGES_REQUESTED** [review]
By itself, the PR is clean and self-descriptive.
1. I'd appreciate if you rebase on cancellation PRs to also take care of their changes.
2. I'd appreciate to see a migration script as well.

File: aqueductcore/backend/services/task_executor.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ async def get_task_by_uuid(
         select(orm.Task)
```
Comment (str-anger) [inline aqueductcore/backend/services/task_executor.py:219]: I cannot comment to line 201, but I assume it needs to be `AsyncResult(db_task.uuid).revoke(terminate=terminate, signal="SIGINT")`

Same for line 202:
`task_info = await _update_task_info(task_id=db_task.uuid, wait=False)`

**Review by str-anger — CHANGES_REQUESTED** [review]
Apart from comments inline, I think class `task.py/TaskProcessExecutionResult` also deserves field renaming. Can you please do this, and propagate this rename to the usage (i.e. method `_update_task_info`)

File: aqueductcore/backend/services/task_executor.py (hunk 1 above)
Comment (jatinriverlane) [reply aqueductcore/backend/services/task_executor.py:219]: Oops, this change was brought by the rebase. Fixed it, thanks.

**Review by str-anger — APPROVED** [review]
Conditional approve, class `TaskProcessExecutionResult` may also benefit from renaming of the field.

**Comment by safeamiiir** [comment]
I found some changes which are missed in the `tests/integration/test_graphql_tasks.py` file:
like lines: `43`, `57`, `88`, ...
I guess they also need to be modified, right?

**Review by safeamiiir — APPROVED** [review]

**Pull request merged**